src/python/pants/rules/core/cloc.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from pathlib import PurePath
```

Comment:
We've always used `Path` instead of `PurePath`. Any particular reason for using the superclass?